item-impl/src/main/java/com/example/auction/item/impl/PItemEntity.java

```diff
@@ -90,15 +113,19 @@ private Behavior completed(PItemState state) {
 
         builder.setReadOnlyCommandHandler(GetItem.class, this::getItem);
 
+        // a completed auctio's item can't be edited.
+        builder.setReadOnlyCommandHandler(UpdateItem.class, (updateItem, ctx) ->
+                ctx.reply(new PUpdateItemResult(UpdateItemResultCodes.CANT_UPDATE_AUCTION_IS_CLOSED))
```

Comment:
Why don't you want an exception sent back? It seems like an exceptional situation.